src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -4373,43 +4373,38 @@ module Queries = struct
           Mina_base.Epoch_seed.to_base58_check
             next_epoch.Mina_base.Epoch_data.Poly.seed
         in
-        let%bind staking_ledger =
-          match Mina_lib.staking_ledger mina with
-          | None ->
-              Deferred.Result.fail "Staking ledger is not initialized."
-          | Some (Genesis_epoch_ledger l) ->
-              return (Ledger.Any_ledger.cast (module Ledger) l)
-          | Some (Ledger_db l) ->
-              return (Ledger.Any_ledger.cast (module Ledger.Db) l)
+        let mina_config = Mina_lib.config mina in
+        let local_state = mina_config.consensus_local_state in
+        let consensus_constants =
+          mina_config.precomputed_values.consensus_constants
+        in
+        let cast_ledger :
+               Consensus.Data.Local_state.Snapshot.Ledger_snapshot.t
+            -> Ledger.Any_ledger.witness = function
+          | Genesis_epoch_ledger l ->
+              Ledger.Any_ledger.cast (module Ledger) l
+          | Ledger_db l ->
+              Ledger.Any_ledger.cast (module Ledger.Db) l
+        in
+        let staking_ledger =
+          Consensus.Hooks.get_epoch_ledger ~constants:consensus_constants
+            ~consensus_state:consensus ~local_state
+          |> cast_ledger
         in
         assert (
           Mina_base.Ledger_hash.equal
             (Ledger.Any_ledger.M.merkle_root staking_ledger)
             staking_epoch.ledger.hash ) ;
-        let%bind next_epoch_ledger =
-          match
-            (* We always want to return the next epoch ledger here, in case we
-               need to hard-fork from a block where it is unfinalized. The
-               safety concern doesn't apply here, because we are only using
-               this to build a snapshot, and never applying it back to the
-               running network.
-            *)
-            Mina_lib.next_epoch_ledger
-              ~unsafe_always_return_ledger_as_if_finalized:true mina
-          with
-          | None ->
-              Deferred.Result.fail "Next epoch ledger is not initialized."
-          | Some `Notfinalized ->
-              failwith "next_epoch_ledger returned a disallowed value"
-          | Some (`Finalized (Genesis_epoch_ledger l)) ->
-              return (Ledger.Any_ledger.cast (module Ledger) l)
-          | Some (`Finalized (Ledger_db l)) ->
-              return (Ledger.Any_ledger.cast (module Ledger.Db) l)
+        let next_epoch_ledger =
+          (* We always want to return the next epoch ledger here, in case we
+             need to hard-fork from a block where it is unfinalized. The
+             safety concern doesn't apply here, because we are only using
+             this to build a snapshot, and never applying it back to the
+             running network.
+          *)
+          Consensus.Data.Local_state.next_epoch_ledger local_state
```

Comment:
Similarly here.

Reply:
Here no block parameters are given, I agree, but essentially the code functions the same way it did before my edit.

Reply:
I am wondering whether it is legit to use this ledger as the "next_epoch_ledger" for that block

Reply:
You didn't change the code but we need some change here.